### DISCUSSION TITLE
add overlay on live streamin

### DISCUSSION THREAD

**Discussion by ways88** [pedroSG94/RootEncoder#1232] [Q&A]
Hi Pedro,

it's possibile to add an overlay meanwhile live is streaming?

i try to explain an example:

when start live I would like to send a message on live on video and the other peaple see with their player.

Best Regards.

**Comment by Ernovation** [comment]
You could use a filter, such as `com.pedro.encoder.input.gl.render.filters.object.TextObjectFilterRender` or `com.pedro.encoder.input.gl.render.filters.object.GifObjectFilterRender`. You could also put the message on a (possibly transparent) AndroidView and use `com.pedro.encoder.input.gl.render.filters.AndroidViewFilterRender`.

**Comment by pedroSG94** [comment]
Hello,

As mentioned by @Ernovation you can use that filters. You have examples in OpenGl activities of app module in the project

**Comment by ways88** [comment]
Thank's a lot, work fine!!!

I have another  2 questions: 

Is it possible to make the overlay scroll horizontally?

in case of the response of the first question, is it possibile to add 2/3 overlay example 1 bottom 1  top and 1 topRigth? 

Best Regards

**Reply by pedroSG94** [reply]
I don't understand what you mean with 2/3 overlay but you can set any view (except a SurfaceView or TextureView) to AndroidviewFilterRender included layouts with children views.

Remember that you need set scale to the filter if the view is not in the correct position. For example you can set scale to 100x100 and then set a layout with size match parent and build your view like a normal xml and all view inside that layout will be streamed

**Reply by ways88** [reply]
Thank you Pedro,

absolutly correct, but for my first question is possible or not?

Best Regards

**Comment by pedroSG94** [comment]
It is possible but could be a bit tricky because you can't handle scroll events in filters directly. For that reason I did this class to detect touch events and move the filter:
https://github.com/pedroSG94/RootEncoder/blob/master/app/src/main/java/com/pedro/streamer/openglexample/OpenGlRtmpActivity.java#L117
You can add a view that scroll and scroll it programatically detecting touch events yourself. For example:
- Add a RecyclerView in a AndroidViewFilterRender
- Add a touch listener to detect touch event detecting scroll in the OpenGlView manually (basically the same that SpriteGestureController but adapted to detect scroll)
- Call scroll method of RecyclerView to scroll depend of the event handled
